kong/runloop/handler.lua

```diff
@@ -1513,7 +1513,14 @@ return {
       -- Report HTTP status for health checks
       local balancer_data = ctx.balancer_data
       if balancer_data and balancer_data.balancer_handle then
-        local status = ngx.status
+        -- https://github.com/nginx/nginx/blob/2485681308bd8d3108da31546cb91bb97813a3fb/src/http/ngx_http_upstream.c#L5656
+        -- because of the way of nginx do the upstream_status variable, it can be
+        -- a string or a number, so we need to parse it to get the status
+        local status = tonumber(sub(ngx.var.upstream_status or "", -3))
```

Comment:
should use already cached `var` to get nginx var value, like `ngx.var.upstream_status`.

More simple code style:
```
local status = tonumber(sub(var.upstream_status or "", -3)) or ngx.status
```